src/NuGet.Core/NuGet.ProjectModel/DependencyGraphSpec.cs

```diff
@@ -371,7 +372,8 @@ private void ParseJson(JObject json)
             var restoreObj = json.GetValue<JObject>("restore");
             if (restoreObj != null)
             {
-                _restore.UnionWith(restoreObj.Properties().Select(prop => prop.Name));
+                _restore.AddRange(restoreObj.Properties().Select(prop => prop.Name)
+                    .ToDictionary(name => name, name => false));
```

Comment:
micro-optimisation: creating a dictionary, just to pass to `AddRange` is a bit of a waste of CPU time. Changing the `Select(prop => prop.Name)` to `Select(props => new KeyValuePair<string, bool>(prop.Name, false)` should have the same result, but without the dictionary needing to hash all the keys, check for duplicates, etc.

Reply:
Ok. Taken the recommendation.